ergotree-interpreter/src/sigma_protocol/fiat_shamir.rs

```diff
@@ -133,6 +135,36 @@ fn fiat_shamir_write_bytes<W: SigmaByteWrite>(
         ProofTreeKind::Conjecture(c) => {
             w.put_u8(INTERNAL_NODE_PREFIX)?;
             w.put_u8(c.conjecture_type() as u8)?;
+            match tree {
```

Comment:
Remove the match statement.

Reply:
Just to confirm, removing the match makes our serialization different to sigma-state - that's correct?

Reply:
Ah, I can't see what happened since the force-push. I forked your PR as of last night (~ 9 hours ago), made these changes, and everything passed.

The match statement I highlighted contained a match arm which had a call to the writer: `w.put_u8(&k)`. The bug I fixed in my code mistakenly emitted 24 extra bytes.

Reply:
(edited previous comment showing what was failing on my side)

Reply:
Damn that sucks, without the `w.put_u8(*k)` included in serialization the signing spec test vector `sig_test_vector_threshold` from `sigma-state` fails. 🤔 

If you give it another shot with the current branch `sigma_protocol::verifier::tests::test_prover_verifier_atleast` will fail, it starts failing when there's [more than 1 proposition threshold bound](https://github.com/ergoplatform/sigma-rust/pull/531/files#diff-5a1347bdbfcb71876cb34a1152319990ca71c461485c5106087a5ac8b3962c3dR406). Haven't had any success figuring it out yet

Reply:
Yeah disregard my suggestion to get rid of that line (you definitely know more than me about this section of the code).

I'll check it out.

Reply:
Aha, got it! Change the `==` to `<=` for this line: https://github.com/ross-weir/sigma-rust/blob/4641814fb535656b72f360e88ea445df7a26d30c/ergotree-ir/src/sigma_protocol/sigma_boolean/cthreshold.rs#L50

Reply:
Thanks so much for the help with this!

Although it makes this test pass it makes a different one fail and the comments in `sigma-state` specify we convert to `AND` if `cur_k == children_left` so I'm not sure this is the way to go